hw/ip/spi_host/data/spi_host.hjson

```diff
@@ -0,0 +1,590 @@
+                   reduce CONFIG.TX_WATERMARK.''',
+          resval: "0x0"
+        },
+        { bits: "25"
+          name: "RXFULL",
+          desc: '''When high, indicates that the receive fifo is full.  Any
+                   ongoing transactions will stall until firmware reads some
+                   from the RXFIFO.'''
+          resval: "0x0"
+        },
+        { bits: "24"
+          name: "RXEMPTY",
+          desc: '''When high, indicates that the receive fifo is empty.
+                   Any reads from RXFIFO will cause an error interrupt.
+                   '''
+          resval: "0x0"
+        },
+        { bits: "23"
+          name: "RXSTALL",
+          desc: '''If high, signifies that an ongoing transaction has stalled
+                   due to lack of available space in the RX FIFO''',
+          resval: "0x0"
+        },
+        { bits: "20",
+          name: "RXWM",
+          desc: '''If high, the amount of data in the RX FIFO now exceeds the
+                   level of CONFIG.RX_WATERMARK words (32b each). To prevent the
+                   reassertion of this interrupt read more data from the RXFIFO, or
+                   increase CONFIG.RX_WATERMARK.''',
+          resval: "0x0"
+        },
+        { bits: "17:9",
+          name: "RXQD",
+          desc: '''Receive queue depth. Indicates how many unread entries are currently in the
+                   RX FIFO.  When active, this result may an underestimate due
+                   to synchronization delays.
+                ''',
+          resval: "0x0"
+        },
+        { bits: "8:0",
+          name: "TXQD",
+          desc: '''Transmit queue depth. Indicates how many unsent entries are currently
+                   in the TXFIFO.  When active, this result may be an
+                   over-estimate, Due to synchronization delays,
+                ''',
+          resval: "0x0"
+        }
+      ]
+    },
+    { name: "CONFIG",
+      desc: '''Configuration register.  Contains fields for enabling the IP
+               .''',
+      swaccess: "rw",
+      hwaccess: "hro",
+      fields: [
+        { bits: "29",
+          name: "CPOL",
+          desc: '''The polarity of the sck clock signal.  When CPOL is 0,
+                   sck is low when idle, and emits high pulses.   When CPOL
+                   is low, sck is high when idle, and emits a series of low
+                   pulses.
+                   '''
+          resval: "0x0"
+        },
+        { bits: "28",
+          name: "FULLCYC",
+          desc: '''Full cycle.  Modifies the CPHA sampling behaviour to allow
+                   for longer device logic setup times.  Rather than sampling the SD
+                   bus a half cycle after shifting out data, the data is sampled
+                   a full cycle after shifting data out.  This means that if
+                   CPHA = 0 data is both shifted the trailing edge, and sampled
```

Comment:
Is there a stray "both" here? I *think* this should be "data is shifted on the trailing edge and sampled a full cycle later".

Reply:
You are correct, the both is a leftover from some previous (inaccurate) wording.

Here's the end result for you:
"This means that if CPHA = 0, data is shifted out on the trailing edge, and sampled a full cycle later.  If CPHA = 1, data is shifted and sampled with the trailing edge, also separated by a full cycle."